core/src/main/java/org/jdbi/v3/core/inference/JdbiInterceptorChain.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.inference;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * An interception chain to process transformation operations.
+ *
+ * @param <S> The source type.
+ * @param <T> The target type.
+ */
+public final class JdbiInterceptorChain<S, T> {
```

Comment:
Are we sure we want to make all this interceptor stuff public? We'll be committing to supporting all of it going forward. If we do expose this interface, we should probably document it.

Reply:
well, it is in a separate package. And the sql array and the mappers are in separate packages. IMHO, just because it is public, it does not have to be part of the public API. I definitely see your point, what would you suggest to do?

Reply:
We reserve all `.internal` packages as not-API. Maybe it could move into an internal package, at least until we are confident to commit to users extending it as well. Or we could mark it as `@Beta` in case we want to reserve the right to change it a bit.